### PR TITLE
Tooltip: Add Even Better Positioning!

### PR DESCRIPTION
**Pull request by DmitriWolf**
## Add Direction Functionality to Tooltip
This PR is an update to @tw15egan 's very good [#392](https://github.com/carbon-design-system/carbon-components-react/pull/392).

- Changes direction proptypes to PropTypes.oneOf(['bottom', 'top', 'left', 'right'])
- Adds in menuOffset prop (optional) to ensure Tooltip is always positioned correctly. Usage is as follows:
```
<Tooltip triggerText="Hover Info Here">
```
or
```
<Tooltip triggerText="Hover Info Here" menuOffset={{left: 10, top: 30}}>
```
Replaces PR [#392](https://github.com/carbon-design-system/carbon-components-react/pull/392)
Fixes Issue [#390](https://github.com/carbon-design-system/carbon-components-react/issues/390).

### CODE REVIEW

**Comment by joshblack** [comment]
Thanks so much @DmitriWolf for taking the time to contribute this! It's incredibly helpful 😄

Am going to add some reviewers to this to try and help it get through, I'd expect some feedback either EOD today or tomorrow. Thanks again!

**Comment by DmitriWolf** [comment]
@tw15egan If you have a moment today, please review this for me.

File: src/internal/FloatingMenu.js [hunk 1]
```diff
@@ -149,6 +172,33 @@ class FloatingMenu extends React.Component {
...
+    adjustedOffsets.top += newOffsets.top || 0;
+    switch (menuDirection) {
+      case 'right':
+        adjustedOffsets.left += 8;
+        adjustedOffsets.top += 4;
+        break;
+      case 'left':
+        adjustedOffsets.top += 4;
+        break;
+    }
```
Comment (DmitriWolf) [inline src/internal/FloatingMenu.js]: I wanted to write this as:
```
switch (menuDirection) {
   case 'right':
     adjustedOffsets.left += 8;
   case 'left':
     adjustedOffsets.top += 4;
 }
```
but your linter wouldn't let me!

Comment (joshblack) [reply src/internal/FloatingMenu.js]: What was the lint rule violation? @DmitriWolf

Comment (DmitriWolf) [reply src/internal/FloatingMenu.js]: `error  Expected a 'break' statement before 'case'  no-fallthrough`

Comment (joshblack) [reply src/internal/FloatingMenu.js]: Definitely will update the lint config to change that, then 😄

**Comment by DmitriWolf** [comment]
@tw15egan please review this for me when you have a moment. Thanks.

**Pull request merged**